src/NHibernate/Transform/QueryAliasToObjectPropertySetter.cs

```diff
@@ -0,0 +1,196 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace NHibernate.Transform
+{
+	/// <summary>
+	/// Resolves setter for an alias with a heuristic: search among properties then fields for matching name and case, then,
+	/// if no matching property or field was found, retry with a case insensitive match. For members having the same name, it
+	/// sorts them by inheritance depth then by visibility from public to private, and takes those ranking first.
+	/// </summary>
+	[Serializable]
```

Comment:
If this class is Serializable we need to implement `IDeserializationCallback` because we have `IDictionary<,>` fields. Or we need to exclude these fields non serialized and add a deserialization constructor for a class to repopulate these fields